api/src/org/labkey/api/dataiterator/SimpleTranslator.java

```
@@ -387,16 +387,32 @@ private Object fetch(Pair<ColumnInfo, Map<?,?>> pair, Object k)
             }
             else
             {
-                if (_allowBulkLoads && _bulkLoads.add(Pair.of(pkCol, pkCol)))
+                if (pkCol.getJdbcType() == JdbcType.GUID)
                 {
-                    TableSelector ts = createSelector(pkCol, pkCol);
-                    ts.forEach(pkCol.getJavaObjectClass(), (Object pk) -> map.put(pk, pk));
+                    if (k == null)
+                        return null;
+                    if (k instanceof String strKey && !GUID.isGUID(strKey))
+                        return null;
                 }
-                else
+
+                try
                 {
-                    TableSelector ts = createSelector(pkCol, pkCol, k);
-                    ts.forEach(pkCol.getJavaObjectClass(), (Object pk) -> map.put(pk, pk));
+                    if (_allowBulkLoads && _bulkLoads.add(Pair.of(pkCol, pkCol)))
+                    {
+                        TableSelector ts = createSelector(pkCol, pkCol);
+                        ts.forEach(pkCol.getJavaObjectClass(), (Object pk) -> map.put(pk, pk));
+                    }
+                    else
+                    {
+                        TableSelector ts = createSelector(pkCol, pkCol, k);
+                        ts.forEach(pkCol.getJavaObjectClass(), (Object pk) -> map.put(pk, pk));
+                    }
                 }
+                catch (Exception e)
+                {
+                    // ignore
```

Where is the exception thrown?  We didn't have a try-catch before.

---

forEach calls rs.next, which throws SQLException

---

Is that really something we want to ignore?

---

It's debatable. This is the call to query by PK, one can argue it's ok for it to silently fail and then the next attempt to query by title will be tried. For my specific GUID/string issue, the check added above should prevent the exception. I've removed the ignore, to minimize the scope of changes.